src/main/java/frc/trigon/robot/subsystems/intake/IntakeConstants.java

```diff
@@ -0,0 +1,122 @@
+package frc.trigon.robot.subsystems.intake;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.Follower;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.system.plant.DCMotor;
+import frc.trigon.robot.constants.RobotConstants;
+import frc.trigon.robot.misc.objectdetectioncamera.SimulationObjectDetectionCameraIO;
+import org.trigon.hardware.misc.simplesensor.SimpleSensor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXMotor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXSignal;
+import org.trigon.hardware.simulation.SimpleMotorSimulation;
+import org.trigon.utilities.mechanisms.SpeedMechanism2d;
+
+import java.util.function.DoubleSupplier;
+
+public class IntakeConstants {
+    private static final int
+            MASTER_MOTOR_ID = 1,
+            FOLLOWER_MOTOR_ID = 2;
+    private static final int DISTANCE_SENSOR_CHANNEL = 0;
+    private static final String
+            MASTER_MOTOR_NAME = "MasterIntakeMotor",
+            FOLLOWER_MOTOR_NAME = "FollowerIntakeMotor",
+            DISTANCE_SENSOR_NAME = "IntakeDistanceSensor";
+    static final TalonFXMotor
+            MASTER_MOTOR = new TalonFXMotor(
+            MASTER_MOTOR_ID,
+            MASTER_MOTOR_NAME,
+            RobotConstants.CANIVORE_NAME
+    ),
+            FOLLOWER_MOTOR = new TalonFXMotor(
+                    FOLLOWER_MOTOR_ID,
+                    FOLLOWER_MOTOR_NAME,
+                    RobotConstants.CANIVORE_NAME
+            );
+    static final SimpleSensor DISTANCE_SENSOR = SimpleSensor.createDutyCycleSensor(DISTANCE_SENSOR_CHANNEL, DISTANCE_SENSOR_NAME);
+
+    private static final InvertedValue
+            MASTER_MOTOR_INVERTED_VALUE = InvertedValue.CounterClockwise_Positive,
+            FOLLOWER_MOTOR_INVERTED_VALUE = InvertedValue.CounterClockwise_Positive;
+    private static final boolean FOLLOWER_MOTOR_OPPOSITE_DIRECTION = false;
+    private static final NeutralModeValue NEUTRAL_MODE_VALUE = NeutralModeValue.Coast;
+    private static final double GEAR_RATIO = 0.003;
+    private static final double
+            DISTANCE_SENSOR_SCALING_SLOPE = 0.0004,
+            DISTANCE_SENSOR_SCALING_INTERCEPT_POINT = 400;
+    static final boolean FOC_ENABLED = true;
+
+    private static final int MOTOR_AMOUNT = 2;
+    private static final DCMotor GEARBOX = DCMotor.getFalcon500Foc(MOTOR_AMOUNT);
+    private static final double MOMENT_OF_INERTIA = 0.003;
+    private static final SimpleMotorSimulation SIMULATION = new SimpleMotorSimulation(
+            GEARBOX,
+            GEAR_RATIO,
+            MOMENT_OF_INERTIA
+    );
+    private static final DoubleSupplier DISTANCE_SENSOR_SIMULATION_VALUE_SUPPLIER = SimulationObjectDetectionCameraIO.HAS_OBJECTS ? () -> 1 : () -> 0;
```

Comment:
this logic is incorrect

Reply:
Then how am I meant to do this?

Reply:
A supplier to the ternary operator, not a ternary operator to the supplier